tests/unit/pipeline/test_filters.py

```diff
@@ -77,3 +78,68 @@ async def filter_record(self, record):
     subject = TestFilter([False, True])
     results = [record async for record in subject.handle_async_record_stream(records())]
     assert_that(results, equal_to([1]))
+
+
+MATCH_INCLUDE_REGEX_CONFIGURATION = [
+    {
+        "value": ">[test]",
+        "include": True,
+        "regex": ".*[\[\]\{\}\(\)\\\/~,]+",  # Matches any string with the characters: []{}()\/~,
+    }
+]
+
+FAILED_MATCH_INCLUDE_REGEX_CONFIGURATION = [
+    {
+        "value": "test",
+        "include": True,
+        "regex": ".*[\[\]\{\}\(\)\\\/~,]+",  # Matches any string with the characters: []{}()\/~,
+    }
+]
+
+MATCH_EXCLUDE_REGEX_CONFIGURATION = [
+    {
+        "value": ">[test]",
+        "include": False,
+        "regex": ".*[\[\]\{\}\(\)\\\/~,]+",  # Matches any string with the characters: []{}()\/~,
+    }
+]
+
+FAILED_MATCH_EXCLUDE_REGEX_CONFIGURATION = [
+    {
+        "value": "test",
+        "include": False,
+        "regex": ".*[\[\]\{\}\(\)\\\/~,]+",  # Matches any string with the characters: []{}()\/~,
+    }
+]
+
+
+@pytest.mark.asyncio
+async def test_match_regex_successful():
+    subject = ValueMatchesRegexFilter.from_file_data(fields=MATCH_INCLUDE_REGEX_CONFIGURATION)
+    result = await subject.filter_record({})
+    assert_that(result, equal_to(False))
+
+
+@pytest.mark.asyncio
+async def test_not_match_regex_successful():
+    subject = ValueMatchesRegexFilter.from_file_data(
+        fields=FAILED_MATCH_INCLUDE_REGEX_CONFIGURATION
+    )
+    result = await subject.filter_record({})
+    assert_that(result, equal_to(True))
+
+
+@pytest.mark.asyncio
+async def test_exclude_match_regex_successful():
+    subject = ValueMatchesRegexFilter.from_file_data(fields=MATCH_EXCLUDE_REGEX_CONFIGURATION)
+    result = await subject.filter_record({})
+    assert_that(result, equal_to(True))
+
+
+@pytest.mark.asyncio
+async def test_exclude_match_regex_failing():
+    subject = ValueMatchesRegexFilter.from_file_data(
+        fields=FAILED_MATCH_EXCLUDE_REGEX_CONFIGURATION
+    )
+    result = await subject.filter_record({})
+    assert_that(result, equal_to(False))
```

Comment:
If we want to DRY this test up, we could [parametrize it](https://github.com/nodestream-proj/nodestream/blob/355cdf677e81f80fa07280060f085d5afdfa1a97/tests/unit/pipeline/normalizers/test_trim_whitespace.py#L7) Not required, just a suggestion